### ISSUE TITLE
VOiceOver focus becomes stuck when the VS Code dialogue changes

### ISSUE THREAD

**Issue by ivanfetch** [microsoft/vscode#88881]
When a new VS Code dialogue box is displayed, such as the `Code` menu -> `About. . .` dialogue, VoiceOver becomes stuck in the `Close`, `Minimize`, and `Zoom` window controls.

1. In VS Code with VoiceOver turned on (Command-f5), verify you can use VoiceOver to navigate UI elements in the primary VS Code window (Ctrl-Opt-left-arrow and Ctrl-Opt-right-arrow).
2. Use VoiceOver to navigate to the menubar (Ctrl-Opt-m), select the `Code` menu, then press Ctrl-Opt-spacebar on the `About Visual Studio Code` menu item.
3. When attempting to navigate the content of the About dialogue, VoiceOver is only able to traverse the `Close`, `Minimize`, and `Zoom` window controls.
4. Toggling VOiceOver off then back on (Cmd-F5) allows the About dialogue to be navigated.
5. Using VoiceOver to activate `OK` in the about dialogue presents the same issue - VoiceOver is stuck in the window controls and is unable to navigate the primary VS Code window.
6. Toggle VoiceOver again to be able to navigate in the primary VS Code window.

[Here is a recorded demo](https://ivanfetch-media.s3.amazonaws.com/github-issues/VS-Code-VoiceOver-Dialogue-Focus.mp4) of this issue.

This does not seem like the same behavior as described in [electron issue 14234](https://github.com/electron/electron/issues/14234), because in this case no underlying dialogue can be seen with VoiceOver.

Version: 1.42.0-insider
Commit: 235695393db763af632fab376b890d63d8a48576
Date: 2020-01-16T18:48:51.004Z
Electron: 7.1.7
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 18.7.0

CC @isidorn

**Comment by isidorn** [comment]
@ivanfetch thanks again for filling this issue.
Just to make sure we are hitting the same issue: after the About dialog is opened in step 3 you can use `Esc` key to close it and normally continue operating VS Code?

Bottom line it seems to be an electron issue. A combintaion of the one I linked and this one https://github.com/electron/electron/issues/12893

**Comment by ivanfetch** [comment]
@isidorn In step 3 (from the first comment), pressing ESCAPE does not get me unstuck from the window controls. I do hear that the dialogue has closed because VoiceOver speaks the title of the edit area, however I am still unable to use VoiceOver keys to navigate beyond the three window controls.

**Comment by isidorn** [comment]
@ivanfetch thanks. Yes, this is an Electron issue. I have contacted our collegue who works on Electron to see if we can prioritise the fix for that issue. Thanks

**Comment by isidorn** [comment]
@yxycode17 has also hit this issue.
@deepak1556 could we look into this from the electron side?

**Comment by yxycode17** [comment]
At times I navigate to those three controls by accident. By any means as long as I enter that region, I can't get rid of it but having to restart VoiceOver.

**Comment by isidorn** [comment]
This should be fixed since @deepak1556 backported an electron fix.
Please let me know if you still face this issue with latest vscode insiders and I will reopen

**Comment by deepak1556** [comment]
The issue is still present for sheet dialogs, opening for further investigation.

**Comment by isidorn** [comment]
Ok, let's reopen then :)

**Comment by isidorn** [comment]
@deepak1556 friendly ping for this? Should we close this?

**Comment by isidorn** [comment]
I can now nicely navigate the about dialog with latest VS Code Insiders and latest BigSur.
@deepak1556 I am not 100% sure where we use sheet dialogs?

Closing since the original issue seems to be fixed
